### ISSUE TITLE
column names have double underscores in them when col_names = F after latest update

### ISSUE THREAD

**Issue by Courvoisier13** [tidyverse/readxl#357]
After #98 was released I am getting "__" in the names of the columns when col_names = F.  https://github.com/tidyverse/readxl/releases it says:
"Default column names for xlsx now start with X__1 instead of X__0. (#98, @zeehio, @krlmlr)"
But before the default column names was X0, X1, X2 ... not X__0, X__1, X__2....
Now, the default column names are X__1, X__2, X__3. I don't understand why there is a double underscore. #98 only discusses fixing the numbers not adding underscores.

Thank you

`> sessionInfo()
R version 3.3.3 (2017-03-06)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readxl_1.0.0

loaded via a namespace (and not attached):
[1] tools_3.3.3      tibble_1.3.0     Rcpp_0.12.10     cellranger_1.1.0`

**Comment by zeehio** [comment]
Reading through the source history:

- The commit https://github.com/tidyverse/readxl/pull/165/commits/370ad6be67868500ed4df1b433352c987485ce89 that I wrote fixed the default column names on xlsx files, before they were "X0, X1..." and after they were "X1, X2...". This is what I wrote on the NEWS file at the time.

- Later on there is commit https://github.com/tidyverse/readxl/commit/a78440d4f4f4fa62f8b89fe5296a262e9cb24180 that changed the column names to "X__1, X__2" (see https://github.com/tidyverse/readxl/commit/a78440d4f4f4fa62f8b89fe5296a262e9cb24180#diff-ab7cb37692316feab1f1692172e9c15aR65 ). The commit message talks about repairing column names like the readr package. I don't know about that, hopefully @jennybc can say more about it. 👍

**Comment by jennybc** [comment]
I am trying to catalyze a uniform tidyverse treatment for column names that are created out of thin air. Discussion here:

https://github.com/tidyverse/tibble/issues/217

So far I have not succeeded, but we have made progress.

Re: readxl: I wanted fabricated column names to stand out more, hence I changed the behaviour intentionally. But I won't say the current form is exactly what I'd like. I'm waiting for this from @krlmlr:

> The next tibble release will have a better `repair_names()`, but probably only after the next dplyr release.

**Comment by krlmlr** [comment]
I'll try to include it in the upcoming tibble release, unless this gives massive revdep failures.

**Comment by jennybc** [comment]
Note to self to revisit thoughts, docs, and tests re: name repair now that tibble has a definitive solution: https://github.com/tidyverse/tibble/pull/248

**Comment by Courvoisier13** [comment]
thank you.

**Comment by jennybc** [comment]
Opening a new issue for future name repair
